det3d/datasets/argoverse/argoverse_common.py

```diff
@@ -0,0 +1,330 @@
+
+    Args:
+        track: (Dict) A dictionary with timestamps (0, 1, 2, ...) as keys and Label as values
+        n_t: (int) Number of timestamps in total
+
+    Returns:
+        interpolated: (Dict) Same as input track, but with gaps filled
+        interpolated_timestamps: (List) List of timestamps of interpolated poses
+    """
+    # TODO: more intelligent interpolation. E.g. we know the orientation of the car and it can't
+    # move along the width axis, so even if there are some small jittery in translation along width axis, shouldn't
+    # keep moving the vehicle along that direction
+    timestamps = list(iter(track))
+    to_interpolate = [t for t in range(n_t) if t not in timestamps]
+
+    existing_boxes = torch.tensor([track[t].boxes for t in timestamps], dtype=torch.float32)
+    existing_boxes_tensor = convert_track_to_tensor(existing_boxes, track)
+
+    if len(to_interpolate) == 0:
+        # Nothing to interpolate
+        return existing_boxes_tensor, []
+    elif len(timestamps) == 1:
+        # Not enough to interpolate. Repeat the tensor n_t times
+        return existing_boxes_tensor.repeat((n_t, 1)), to_interpolate
+
+    # Fit rotations
+    rots = Rotation.from_euler('x', existing_boxes_tensor[:, 6])
+    spline = RotationSpline(timestamps, rots)
+    interpolated_rots = spline(to_interpolate)
+
+    # Fit x, y and z of translation separately
+    # UnivariateSpline works the best practically in our case, but it doesn't work with <= 3 points
+    interp_cls = interpolate.UnivariateSpline if len(timestamps) > 3 else interpolate.KroghInterpolator
+    func_x = interp_cls(timestamps, existing_boxes_tensor[:, 0])
+    func_y = interp_cls(timestamps, existing_boxes_tensor[:, 1])
+    func_z = interp_cls(timestamps, existing_boxes_tensor[:, 2])
+
+    interpolated_x = func_x(to_interpolate)
+    interpolated_y = func_y(to_interpolate)
+    interpolated_z = func_z(to_interpolate)
+
+    interpolated = torch.zeros((n_t, 7))  # 7 values per bounding box
+
+    interpolated[timestamps, :] = existing_boxes_tensor
+
+    first_label = track[list(iter(track))[0]]
+
+    interpolated[to_interpolate, 0] = torch.tensor(interpolated_x, dtype=torch.float32)
+    interpolated[to_interpolate, 1] = torch.tensor(interpolated_y, dtype=torch.float32)
+    interpolated[to_interpolate, 2] = torch.tensor(interpolated_z, dtype=torch.float32)
+
+    # Length, width, height and label are set to the same value as the first given label
+    interpolated[to_interpolate, 3] = first_label.record.length
+    interpolated[to_interpolate, 4] = first_label.record.width
+    interpolated[to_interpolate, 5] = first_label.record.height
+
+    interpolated[to_interpolate, 6] = torch.tensor(interpolated_rots.as_euler('xyz')[:, 0], dtype=torch.float32)
+
+    return interpolated, to_interpolate
+
+def convert_track_to_tensor(bboxes, track):
+    center_and_yaws = find_center_and_rotation(bboxes)
+    # TODO: centerpoint expects wlh?
+    l_w_h = torch.tensor([
```

Comment:
Does this run even though Centerpoint doesn't accept it?

Reply:
This is fine. If we put in lwh, we get lwh out. As long as we interpret them as lwh, it's fine.